tfjs-tflite-node-codelab/webnn_inference_working/renderer.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+// CODELAB part 1: Import tfjs-tflite-node here.
+const {loadTFLiteModel} = require('tfjs-tflite-node');
+const tf = require('@tensorflow/tfjs');
+// CODELAB part 2: Import the delegate here.
+const {WebNNDelegate} = require('webnn-tflite-delegate');
+const fs = require('fs');
+const Stats = require('stats.js');
+
+async function getWebcam() {
+  const webcam = document.createElement('video');
+  webcam.width = 224;
+  webcam.height = 224;
+  const stream = await navigator.mediaDevices.getUserMedia({
+    audio: false,
+    video: {
+        mandatory: {
+          minHeight: webcam.height,
+          minWidth: webcam.width,
+          maxHeight: webcam.height,
+          maxWidth: webcam.width,
+        }
+    }
+  });
+  // Flip the image so the video looks like a mirror.
+  webcam.style.transform = 'scaleX(-1)';
+  webcam.srcObject = stream;
+  webcam.play();
+  return webcam;
+}
+
+async function main() {
+  const stats = new Stats();
+  document.body.appendChild(stats.dom);
+
+  const webcam = await getWebcam();
+  document.body.appendChild(webcam);
+
+  // For displaying results
+  const resultHeader = document.createElement('h1');
+  resultHeader.innerHTML = 'Classification: ';
+  document.body.appendChild(resultHeader);
+
+  const resultList = document.createElement('ul');
+  document.body.appendChild(resultList);
+
+  function showPrediction(data, labels) {
+    resultList.innerHTML = ''; // Clear the list
+    for (let i = 0; i < data.length; i++) {
+      const percent = data[i].toFixed(1);
+      const resultText = `${labels[i]}: ${percent}`;
+      const result = document.createElement('li');
+      result.innerHTML = resultText;
+      resultList.appendChild(result);
+    }
+  }
+
+  // CODELAB part 1: Load the model here.
+  const modelPath = './model/model_unquant.tflite';
+  const cpuModel = await loadTFLiteModel(modelPath);
+  const labels = fs.readFileSync('./model/labels.txt', 'utf8')
+        .split(/\r?\n/);
+
+  // CODELAB part 2: Load the delegate model here.
+  let webnnModel = await loadTFLiteModel(modelPath, {
+    delegates: [new WebNNDelegate([['webnn_device', '0']])],
```

Comment:
The line here that registers the delegate will need to change now that #26 is merged, but I don't think it's a very big change.